src/object/srv_obj.c

```diff
@@ -819,6 +819,12 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 	dispatch = update && orw->orw_shard_tgts.ca_arrays != NULL;
 	tag = dss_get_module_info()->dmi_tgt_id;
 
+	/* FIXME: until distributed transaction. */
+	if (orw->orw_epoch == DAOS_EPOCH_MAX) {
+		orw->orw_epoch = daos_ts2epoch();
+		D_DEBUG(DB_IO, "overwrite epoch "DF_U64"\n", orw->orw_epoch);
+	}
```

Comment:
I don't think that this is the right place to do this check. We should do it at least after the container handle validation and and pool map version check.

I will let fanyong and xuezhao comment further.

Reply:
the suggestion looks good to me, probably can do it before or inside ds_obj_req_disp_prepare().